edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookPermissionServiceImpl.java

```diff
@@ -1178,6 +1179,59 @@ public List getViewableStudentsForUser(String gradebookUid, String userId, List
 		
 	}
 
+	/**
+	 * Get a list of permissions defined for the given user based on section and role or all sections if allowed. 
+	 * This method checks realms permissions for role/section and is independent of the 
+	 * gb_permissions_t permissions.
+	 *
+	 * note: If user has the grade privilege, they are given the GraderPermission.VIEW_COURSE_GRADE permission to match
+	 * GB classic functionality. This needs to be reviewed.
+	 *
+	 * @param userUuid
+	 * @param siteId
+	 * @param role user Role
+	 * @return list of {@link org.sakaiproject.service.gradebook.shared.PermissionDefinition PermissionDefinitions} or empty list if none
+	 */
+	public List<PermissionDefinition> getRealmsPermissionsForUser(String userUuid,String siteId, Role role){
+
+		List<PermissionDefinition> permissions = new ArrayList<PermissionDefinition>();
+
+		if( this.getGradebookService().isUserAllowedToGrade(siteId,userUuid)){
+			//FIXME:giving them view course grade (this needs to be reviewed!!), 
+			//it appears in GB classic, User can view course grades if they have the ability to grade in realms
```

Comment:
Yeah viewing course grades is a new permission in GBNG that can be given to TA's. @kyleblythe I cannot remember the rationale for this, do we have doco? Should have created feature files.